### ISSUE TITLE
Ensure that dashboard pages are all responsive

### ISSUE THREAD

**Issue by jonathancrabtree** [nih-cfde/dashboard#79]
Exact requirements may need to be refined, but at minimum each of the current dashboard pages (index.html, summary_graph.html, dcc_review.html, and dashboard.html) and also the new personalized dashboard page, should resize gracefully down to at least iPad (CSS not pixel) resolution of 768 x 1024. Related cases include #43, #73, #76, and there was some previous discussion in #66.

**Comment by jonathancrabtree** [comment]
I've taken a first pass at doing this for the landing page (index.html), although the changes won't show up in app-dev until after tonight's nightly update of app-dev. I've also changed the look of the layout somewhat, though nothing drastic.  Here are some side-by-side comparisons of the current (app.nih-cfde.org) and proposed landing pages at different horizontal resolutions. 

Super-wide (2305 pixels) current:
![Screen Shot 2021-04-13 at 5 03 11 PM](https://user-images.githubusercontent.com/5394345/114621169-5f13ac00-9c7a-11eb-96f2-513a92aee484.png)

Super-wide (2305 pixels) new. Note the new location of the title at the top left and summary box at the top right. Content is allowed to expand more than in the current version but we could place a lower limit on the width:
![Screen Shot 2021-04-13 at 5 03 48 PM](https://user-images.githubusercontent.com/5394345/114621194-66d35080-9c7a-11eb-9d7a-f7036d58afee.png)

Regular (1300 pixels) current:
![Screen Shot 2021-04-13 at 5 08 53 PM](https://user-images.githubusercontent.com/5394345/114621749-14466400-9c7b-11eb-9d83-122441cd7e85.png)

Regular (1300 pixels) new (very similar):
![Screen Shot 2021-04-13 at 5 09 14 PM](https://user-images.githubusercontent.com/5394345/114621763-18728180-9c7b-11eb-96a2-911dd70cc948.png)

Somewhat narrow (1155 pixels) current. Content is now offscreen to the right. Good example of neo-classical "Let them eat scrollbar" school of web design:
![Screen Shot 2021-04-13 at 5 11 38 PM](https://user-images.githubusercontent.com/5394345/114622311-cb42df80-9c7b-11eb-8f5e-f8f3c390acd6.png)

Somewhat narrow (1155 pixels) new:
![Screen Shot 2021-04-13 at 5 11 57 PM](https://user-images.githubusercontent.com/5394345/114622337-d3028400-9c7b-11eb-9eb7-08ada2283f29.png)

Narrow (1024 pixels) current:
![Screen Shot 2021-04-13 at 5 16 00 PM](https://user-images.githubusercontent.com/5394345/114622539-152bc580-9c7c-11eb-8ef2-91fc6174df16.png)

Narrow (1024 pixels) new. "Funding" hops onto  a new line:
![Screen Shot 2021-04-13 at 5 16 23 PM](https://user-images.githubusercontent.com/5394345/114622559-1c52d380-9c7c-11eb-91eb-0562b231f362.png)

Very narrow (800 pixels) current. Switching to Developer Tools view:
![Screen Shot 2021-04-13 at 5 19 08 PM](https://user-images.githubusercontent.com/5394345/114622828-79e72000-9c7c-11eb-8562-4a652ba79e96.png)

Very narrow (800 pixels) new. Note the navbar is a bit funky, as is the black bar at the bottom of the page (not visible in this image, but shown in the previous one - that blank area at the bottom right seems to be connected to the navbar somehow):
![Screen Shot 2021-04-13 at 5 19 22 PM](https://user-images.githubusercontent.com/5394345/114622890-8ff4e080-9c7c-11eb-8b36-e4e20ee9f6a5.png)

**Comment by jonathancrabtree** [comment]
768 pixels (current):
![Screen Shot 2021-04-13 at 5 22 41 PM](https://user-images.githubusercontent.com/5394345/114623205-eeba5a00-9c7c-11eb-9655-8291d258b427.png)

768 pixels (new):
![Screen Shot 2021-04-13 at 5 22 57 PM](https://user-images.githubusercontent.com/5394345/114623218-f417a480-9c7c-11eb-9a95-f6e9559086ab.png)

767 pixels (new).  767 pixels is when the navbar switches over to its compact mode:
![Screen Shot 2021-04-13 at 5 24 17 PM](https://user-images.githubusercontent.com/5394345/114623374-2e814180-9c7d-11eb-8289-7a1ab2c48d89.png)

500 pixels (current):
![Screen Shot 2021-04-13 at 5 25 41 PM](https://user-images.githubusercontent.com/5394345/114623495-5a9cc280-9c7d-11eb-868a-011e8e2a3ae2.png)

500 pixels (new):
![Screen Shot 2021-04-13 at 5 26 42 PM](https://user-images.githubusercontent.com/5394345/114623568-6ee0bf80-9c7d-11eb-9d08-937141d8df7e.png)

250 pixels (new). Note the issue with the navbar and the fact that the artifact in the page footer seems to correspond to whatever's going on with the navbar. Will have to check if this shows up on app-dev tomorrow as I may have older navbar code/CSS on my dev server:
![Screen Shot 2021-04-13 at 5 27 26 PM](https://user-images.githubusercontent.com/5394345/114623731-a9e2f300-9c7d-11eb-96c3-24f5f5547fca.png)
![Screen Shot 2021-04-13 at 5 27 34 PM](https://user-images.githubusercontent.com/5394345/114623739-ae0f1080-9c7d-11eb-9bc9-fbe8df8726e5.png)

**Comment by jonathancrabtree** [comment]
@jrchudy once this update is on app-dev (tomorrow, I assume) I'll have to check and see if the navbar (and footer) problems are still there and if they are then maybe you can investigate. That white blank space to the right of the navbar appears to be correlated with the white blank space on the right side of the (black) page footer. It varies depending on page width and can be very small ranging up to almost half the width of the screen, as shown above. Also, once everyone is happy with the layout on index.html I can do essentially the same thing on the other dashboard pages.

**Comment by jrchudy** [comment]
I'll have to look at this tomorrow as well since this is likely an issue with the styles applied to the navbar.

**Comment by jonathancrabtree** [comment]
app-dev looks consistent with my dev site. Current state:

- index.html - using new layout, looks good except for navbar at very narrow screen sizes
- summary_graph.html - no known layout issues
- dashboard.html - not yet updated to new layout
- dcc_review.html - not yet updated to new layout

**Comment by jrchudy** [comment]
Some updates from testing the same screen sizes that Jonathan did in the past
 - width = 2305 px
    - The udashbaord, pdashboard, and dcc_review don't stretch the page content to fill the full width
    - The title on dcc_review pulls left, un-aligning it from the rest of the content
 - width = 1300 px
    - no changes from earlier assessment, pages look good
 - width = 1155 px
    - dcc_review does not shrink the content to adjust for the smaller screen
 - width = 1024 px
    - same issue with dcc_review page
 - width = 800 px
    - weird white space on right side of footer starts showing for home and dashboard pages
    - udashboard
       - alert has the "x" move to next line
       - carousel doesn't wrap or internally scroll
       - download button for chart is out of the box
    - content on the pdashboard is starting to look "slim" but not unreadable/un-useable yet
    - continued issue with dcc_review page
 - width = 768 px
    - same issues as 800px
 - width = 767 px
    - navbar switches to compact mode, footer issue is less noticeable but still there
    - udashboard has the x/y axis dropdowns for the plot stack now with the download button overlaying the x-axis dropdown
    - pdashboard switches to stacking saved searches, then favorites, then the plot
    - continued issue with dcc_review page
 - width = 500 px
    - banner text is truncated
    - footer issue is still there
    - continued issue with dcc_review page
 - width = 250 px
    - same issues Jonathan noted before with navbar/footer being way worse
    
One thing to note, part of the application (I think maybe the navbar since this applies to deriva pages too) limits the "minimum" screen width when stretching/shrinking the browser width. The width can't "normally" go below ~538px so the issues specific to 500px and 250px are much less likely to occur when using a desktop.

I did test on mobile (not sure how to check the "pixel width of my mobile display" and do see something similar to the 500px and 250px cases, although the application is not designed to be "completely" mobile friendly.


To summarize similar to how Jonathan did before:
Current state of app-dev (Dec 16th, 2021):
 - index.html - looks good except for navbar and banner text at very narrow screen sizes
 - summary_graph.html - no known layout issues
 - udashboard.html - content does not shrink as the page width gets smaller
 - pdashboard.html - looks very good
 - dcc_review.html - content does not shrink as the page width gets smaller (probably never got updated to new layout)

**Comment by jrchudy** [comment]
Need to test in safari, firefox, and IE/edge.
